citrination_client/models/client.py

```diff
@@ -0,0 +1,95 @@
+from citrination_client.base.base_client import BaseClient
+from citrination_client.models import *
+
+from citrination_client.models import routes as routes
+
+class ModelsClient(BaseClient):
+    """
+    A client that encapsulates interactions with models on Citrination.
+    """
+
+    def __init__(self, api_key, webserver_host="https://citrination.com", suppress_warnings=False):
+        members = [
+            "tsne",
+            "predict"
+        ]
+        super(ModelsClient, self).__init__(api_key, webserver_host, members, suppress_warnings=suppress_warnings)
+
+    def tsne(self, data_view_id):
+        """
+        Get the t-SNE projection, including responses and tags.
+
+        :param data_view_id: The ID of the data view to retrieve TSNE from
+        :type data_view_id: int
+        :return: The TSNE analysis
+        :rtype: :class:`Tsne`
+        """
+        analysis = self._data_analysis(data_view_id)
+        projections = analysis['projections']
+        tsne = Tsne()
+        for k, v in projections.items():
+            projection = Projection(
+                    xs=v['x'],
+                    ys=v['y'],
+                    responses=v['label'],
+                    tags=v['inputs'],
+                    uids=v['uid']
+                )
+            tsne.add_projection(k, projection)
+
+        return tsne
+
+    def predict(self, data_view_id, candidates, method="scalar", use_prior=True):
+        """
+        Predict endpoint
+
+        :param data_view_id: The ID of the data view to use for prediction
+        :type data_view_id: str
+        :param candidates: A list of candidates to make predictions on
+        :type candidates: list of dicts
+        :param method: Method for propagating predictions through model
+            graphs
+        :type method: str ("scalar" or "from_distribution")
+        :param use_prior:  Whether to apply prior values implied by the property descriptors
+        :type use_prior: bool
+        :return: The result of the prediction
+        :rtype: :class:`PredictionResult`
```

Comment:
This should return a list of `PredictionResult`, one for each input candidate.

Reply:
👍